src/filesystem/index.ts

```diff
@@ -577,6 +716,349 @@ server.setRequestHandler(CallToolRequestSchema, async (request) => {
   }
 });
 
+// Add this handler after the other resource handlers
+server.setRequestHandler(ListResourceTemplatesRequestSchema, async () => {
+  return {
+    resourceTemplates: [
+      {
+        uriTemplate: "file://{path}",
+        name: "File Resource",
+        description: "A file on the local filesystem",
+      },
+      {
+        uriTemplate: "folder://{path}",
+        name: "File Resource",
+        description: "A folder on the local filesystem",
+      },
+    ],
+  };
+});
+
+server.setRequestHandler(ReadResourceRequestSchema, async (request) => {
+  try {
+    // Handle both file:// and folder:// URIs
+    const match = request.params.uri.match(/^(file|folder):\/\/(.+)/);
+    if (!match) {
+      throw new Error('Invalid URI format');
+    }
+
+    const [, scheme, pathStr] = match;
```

Comment:
Consider using the `URL` class for parsing here–makes it easier to check the scheme and extract different parts of the URL.

Reply:
great point!